WordPress/Sniffs/WP/I18nSniff.php

```diff
@@ -622,12 +624,30 @@ protected function check_text( $context ) {
 		 *
 		 * Strip placeholders and surrounding quotes.
 		 */
-		$non_placeholder_content = trim( $this->strip_quotes( $content ) );
-		$non_placeholder_content = preg_replace( self::SPRINTF_PLACEHOLDER_REGEX, '', $non_placeholder_content );
+		$content_without_surrounding_quotes = trim( $this->strip_quotes( $content ) );
+		$non_placeholder_content = preg_replace( self::SPRINTF_PLACEHOLDER_REGEX, '', $content_without_surrounding_quotes );
 
 		if ( '' === $non_placeholder_content ) {
 			$this->phpcsFile->addError( 'Strings should have translatable content', $stack_ptr, 'NoEmptyStrings' );
 		}
+
+		/*
+		 * NoHtmlWrappedStrings.
+		 *
+		 * Strip surrounding quotes.
+		 */
+		if ( preg_match( self::WRAPPED_IN_HTML_REGEX, $content_without_surrounding_quotes, $tags ) ) {
+			// Starts with < and ends with >
+			$starting_tag = substr( $tags[1], 1, -1 );
+			// Starts with </ and ends with >
+			$ending_tag = substr( $tags[2], 2, -1 );
+
+			// Don't wrap in tags if the starting and ending tag are different
+			if ( $starting_tag === $ending_tag ) {
+				// Wrapped in HTML.
+				$this->phpcsFile->addError( 'Strings should not be wrapped in HTML', $stack_ptr, 'NoHtmlWrappedStrings' );
```

Comment:
IMO a `warning` would be more appropriate.

Reply:
101ae2bc